hardware_integration/include/isobus/hardware_integration/can_hardware_interface.hpp

```diff
@@ -84,88 +83,114 @@ namespace isobus
-		static isobus::EventDispatcher<> &get_periodic_update_event_dispatcher();
+		static EventDispatcher<> &get_periodic_update_event_dispatcher();
+
+		/// @brief Call this periodically if you have threads disabled.
+		/// @note Try to call this very often, say at least every millisecond to ensure CAN messages are retrieved from the hardware
+		static void update();
 
-		/// @brief Set the interval between periodic updates
+		/// @brief Set the interval between periodic updates to the network manager
 		/// @param[in] value The interval between update calls in milliseconds
 		static void set_periodic_update_interval(std::uint32_t value);
 
-		/// @brief Get the interval between periodic updates
+		/// @brief Get the interval between periodic updates to the network manager
 		/// @returns The interval between update calls in milliseconds
 		static std::uint32_t get_periodic_update_interval();
 
 	private:
-		/// @brief Stores the Tx/Rx queues, mutexes, and driver needed to run a single CAN channel
-		struct CANHardware
+		/// @brief Stores the data for a single CAN channel
+		class CANHardware
 		{
-			std::mutex messagesToBeTransmittedMutex; ///< Mutex to protect the Tx queue
-			std::deque<isobus::CANMessageFrame> messagesToBeTransmitted; ///< Tx message queue for a CAN channel
+		public:
+			/// @brief Constructor for the CANHardware
+			/// @param[in] queueCapacity The capacity of the transmit and receive queues
+			explicit CANHardware(std::size_t queueCapacity);
+
+			/// @brief Destructor for the CANHardware
+			virtual ~CANHardware();
 
-			std::mutex receivedMessagesMutex; ///< Mutex to protect the Rx queue
-			std::deque<isobus::CANMessageFrame> receivedMessages; ///< Rx message queue for a CAN channel
+			/// @brief Starts this hardware channel
+			/// @returns `true` if the channel was started, otherwise `false`
+			bool start();
+
+			/// @brief Stops this hardware channel
+			/// @returns `true` if the channel was stopped, otherwise `false`
+			bool stop();
+
+			/// @brief Try to transmit the frame to the hardware
+			/// @param[in] frame The frame to transmit
+			/// @returns `true` if the frame was transmitted, otherwise `false`
+			bool transmit_can_frame(const CANMessageFrame &frame) const;
+
+			/// @brief Receives a frame from the hardware and adds it to the receive queue
+			/// @returns `true` if a frame was received, otherwise `false`
+			bool receive_can_frame();
+
+#if !defined CAN_STACK_DISABLE_THREADS && !defined ARDUINO
+			/// @brief Starts the receiving thread for this CAN channel
+			void start_threads();
+
+			/// @brief Stops the receiving thread for this CAN channel
+			void stop_threads();
+
+			/// @brief The receiving thread loop for this CAN channel
+			void receive_thread_function();
 
 			std::unique_ptr<std::thread> receiveMessageThread; ///< Thread to manage getting messages from a CAN channel
+			bool receiveThreadRunning; ///< Flag to indicate if the receive thread is running
```

Comment:
MSVC complains that this is not always initialized. Might want a `= false` on there